metamorph/store/dynamodb/dynamodb_test.go

```diff
@@ -0,0 +1,204 @@
+package dynamodb
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/libsv/go-bt/v2"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/ory/dockertest"
+	"github.com/stretchr/testify/require"
+
+	"github.com/bitcoin-sv/arc/metamorph/metamorph_api"
+	"github.com/bitcoin-sv/arc/metamorph/store"
+)
+
+const (
+	hostname = "test-host"
+)
+
+var (
+	TX1Raw         = "01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff1a0386c40b2f7461616c2e636f6d2f00cf47ad9c7af83836000000ffffffff0117564425000000001976a914522cf9e7626d9bd8729e5a1398ece40dad1b6a2f88ac00000000"
+	TX1RawBytes, _ = hex.DecodeString(TX1Raw)
+	TX1, _         = bt.NewTxFromBytes(TX1RawBytes)
+	TX1Hash, _     = chainhash.NewHashFromStr(TX1.TxID())
+
+	TX2raw         = "010000000000000000ef016f8828b2d3f8085561d0b4ff6f5d17c269206fa3d32bcd3b22e26ce659ed12e7000000006b483045022100d3649d120249a09af44b4673eecec873109a3e120b9610b78858087fb225c9b9022037f16999b7a4fecdd9f47ebdc44abd74567a18940c37e1481ab0fe84d62152e4412102f87ce69f6ba5444aed49c34470041189c1e1060acd99341959c0594002c61bf0ffffffffe7030000000000001976a914c2b6fd4319122b9b5156a2a0060d19864c24f49a88ac01e7030000000000001976a914c2b6fd4319122b9b5156a2a0060d19864c24f49a88ac00000000"
+	TX2RawBytes, _ = hex.DecodeString(TX2raw)
+	TX2, _         = bt.NewTxFromBytes(TX2RawBytes)
+	TX2Hash, _     = chainhash.NewHashFromStr(TX2.TxID())
+)
+
+func NewDynamoDBIntegrationTestRepo(t *testing.T) (*DynamoDB, *dynamodb.Client) {
+	t.Helper()
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	resource, err := pool.Run("amazon/dynamodb-local", "latest", []string{})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err := pool.Purge(resource)
+		require.NoError(t, err)
+	})
+
+	resolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+		return aws.Endpoint{
+			PartitionID:   "aws",
+			URL:           "http://localhost:" + resource.GetPort("8000/tcp"),
```

Comment:
maybe worth extracting this as const? 
it will be hard to maintain this tests in docker setup this way.